plugins/bazaar/src/components/SortView/SortView.tsx

```diff
@@ -0,0 +1,163 @@
+  const [catalogEntities, setCatalogEntities] = useState<Entity[]>([]);
+  const [bazaarMembers, setBazaarMembers] = useState<Map<EntityRef, number>>(
+    new Map(),
+  );
+  const [bazaarProjects, setBazaarProjects] = useState<BazaarProject[]>([]);
+  const bazaarApi = useApi(bazaarApiRef);
+  const catalogApi = useApi(catalogApiRef);
+
+  const compareProjectsByDate = (
+    a: BazaarProject,
+    b: BazaarProject,
+  ): number => {
+    const dateA = new Date(a.updatedAt).getTime();
+    const dateB = new Date(b.updatedAt).getTime();
+    return dateB - dateA;
+  };
+
+  const handleCloseNoProjects = () => {
+    setOpenNoProjects(false);
+  };
+
+  const { loading } = useAsync(async () => {
+    const entities = await catalogApi.getEntities({
+      filter: {
+        kind: 'Component',
+        'metadata.annotations.backstage.io/edit-url': CATALOG_FILTER_EXISTS,
+      },
+      fields: ['apiVersion', 'kind', 'metadata', 'spec'],
+    });
+
+    const response = await bazaarApi.getEntities();
+    const dbProjects: BazaarProject[] = [];
+    const bazaarProjectRefs: string[] = [];
+
+    response.data.forEach((project: any) => {
+      dbProjects.push({
+        entityRef: project.entity_ref,
+        name: project.name,
+        status: project.status,
+        announcement: project.announcement,
+        community: project.community,
+        updatedAt: project.updated_at,
+      });
+
+      bazaarProjectRefs.push(project.entity_ref);
+    });
+
+    setBazaarMembers(await bazaarApi.getMemberCounts(dbProjects));
+    setBazaarProjects(dbProjects);
+    setCatalogEntities(
+      entities.items.filter((entity: Entity) => {
+        return !bazaarProjectRefs.includes(stringifyEntityRef(entity));
+      }),
+    );
+  });
+
+  if (loading) {
+    return <Progress />;
+  }
+
+  return (
+    <Content noPadding>
+      <AlertBanner
+        open={openNoProjects}
+        message={
+          <div>
+            No project available. Please{' '}
+            <Link
+              style={{ color: 'inherit', fontWeight: 'bold' }}
+              href="/create"
+            >
+              create a project
+            </Link>{' '}
+            from a template first.
+          </div>
+        }
+        handleClose={handleCloseNoProjects}
+      />
+      <ContentHeader title="Latest updated">
+        <Button
+          variant="contained"
+          color="primary"
+          onClick={() => {
+            if (catalogEntities.length !== 0) {
+              setOpenAdd(true);
+            } else {
+              setOpenNoProjects(true);
+            }
+          }}
+        >
+          Add project
+        </Button>
+        <AddProjectDialog
+          catalogEntities={catalogEntities}
+          handleClose={() => {
+            setOpenAdd(false);
+          }}
+          open={openAdd}
+          setBazaarProjects={setBazaarProjects}
+          setCatalogEntities={setCatalogEntities}
```

Comment:
fyi react will get angry when you call setters of one component from inside another component; probably better if the dialog had a `onAdded` callback which this component could react to by refreshing itself (for example using a `useAsyncFn` for the async work and retriggering it). Not a huge deal but might be good to know